.github/workflows/makefile.yml

```diff
@@ -0,0 +1,280 @@
+name: Makefile CI
+
+on:
+  push:
+  pull_request:
+env:
+  # BASIC_PACKAGES is used fo the basic tools (make)
+  BASIC_PACKAGES: git build-essential ccache libgc-dev graphviz libunwind-dev libreadline-dev pkg-config
+  # MORE_PACKAGES is used for more tools and libs (make more)
+  MORE_PACKAGES: libgmp-dev libcurl4-openssl-dev libevent-dev openjdk-17-jdk-headless libgles-dev libegl-dev libsdl1.2-dev libsdl2-dev libsdl2-mixer-dev libsdl2-gfx-dev libsdl2-image-dev libsdl2-net-dev libsdl2-ttf-dev libgtk-3-dev inkscape postgresql libxdg-basedir-dev gettext libsqlite3-dev libpq-dev libmongoc-dev openmpi-bin libopenmpi-dev
+  JAVA_HOME: /usr/lib/jvm/java-17-openjdk-amd64/
+  JNI_LIB_PATH: /usr/lib/jvm/java-17-openjdk-amd64/lib/server/
+  LD_LIBRARY_PATH: /usr/lib/jvm/java-17-openjdk-amd64/lib/server/
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y $BASIC_PACKAGES
+    - run: make
+    - uses: actions/upload-artifact@v4
+      with:
+        path: bin
+        name: bin
+
+  sanity-check:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - run: misc/jenkins/ensuregit.sh
+    - run: misc/jenkins/checkwhitespaces.sh
+      if: success() || failure()
+    - run: misc/jenkins/checklicense.sh
+      if: success() || failure()
+    - run: misc/jenkins/checksignedoffby.sh
+      if: success() || failure()
+
+  tests-some:
+    runs-on: ubuntu-24.04
+    needs: [build]
+    container: "ubuntu:24.04"
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+      mongo:
+        image: mongo
+      neo4j:
+        image: 'neo4j:2.3'
```

Comment:
Do we want to open an issue if higher version of neo4j causes a regression?

Reply:
This is a good point. We might, but this is not the responsibility of an unrelated PR that run some CI to deal with such breakage.
Maybe having another workflow on master that run periodically in order to detect regression with external components.